### PR TITLE
Fix: Flatland for Noetic

### LINKED ISSUE #316: flatland-server fails to build in ROS Iron
Trying to use `ros-iron-flatland` package results in the following error:

```sh
error: builder for '/nix/store/qmsv1gf5ahjrgliwljjg8cdb6074bdn5-ros-noetic-flatland-msgs-1.3.3-r1.drv' failed with exit code 1;
       last 10 log lines:
       > -- Using these message generators: gencpp;geneus;genlisp;gennodejs;genpy
       > CMake Error at /nix/store/wjbqjwmnbi24qsvfh589my84fjr8wa3l-cmake-3.26.4/share/cmake-3.26/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
       >   Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE)
       > Call Stack (most recent call first):
       >   /nix/store/wjbqjwmnbi24qsvfh589my84fjr8wa3l-cmake-3.26.4/share/cmake-3.26/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
       >   /nix/store/wjbqjwmnbi24qsvfh589my84fjr8wa3l-cmake-3.26.4/share/cmake-3.26/Modules/FindPkgConfig.cmake:99 (find_package_handle_standard_args)
       >   CMakeLists.txt:17 (find_package)
```

I'm using the following `flake.nix`:

```nix
{
  inputs = {
    nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
    flake-utils.url = "github:numtide/flake-utils";
    ros-overlay.url = "github:lopsided98/nix-ros-overlay";
  };

  outputs = { nixpkgs, flake-utils, ros-overlay, ... }:
    with nixpkgs.lib;
    with flake-utils.lib;

    eachSystem allSystems (system:
      let
        pkgs = import nixpkgs {
          inherit system;
          overlays = [ ros-overlay.overlays.default ];
        };
      in
      {
        devShells.default = with pkgs; mkShell {
          buildInputs = with rosPackages.noetic; [
            (buildEnv {
              paths = [
                catkin
                cmake-modules
                geometry-msgs
                interactive-markers
                flatland-server
                opencv
                pluginlib
                roslaunch
                rostest
                rviz
                std-msgs
                tf
                tf2
                tf2-geometry-msgs
                visualization-msgs
                yaml-cpp
              ];
            })
          ];
        };
      });
}

```

### PR DESCRIPTION
**Pull request by Educorreia932**
Fixes #316

### CODE REVIEW

File: distros/noetic/flatland-server/yaml_preprocessor.patch [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+@@ -47,12 +47,15 @@
+ #ifndef FLATLAND_SERVER_YAML_PREPROCESSOR_H
+ #define FLATLAND_SERVER_YAML_PREPROCESSOR_H
+
++extern "C"
++{
++    #include <lua.h>
++    #include <lualib.h>
++    #include <lauxlib.h>
++}
```
Comment (lopsided98) [inline distros/noetic/flatland-server/yaml_preprocessor.patch:12]: Why isn't this necessary on Ubuntu? Does the Ubuntu package really compile Lua as C++?

**Review by lopsided98 — CHANGES_REQUESTED** [review]
You shouldn't manually edit the files in `distro/noetic` (except for `overrides.nix`) because they are automatically generated. Instead, make any necessary changes as overrides in `distros/noetic/overrides.nix` or `distros/ros1-overlay.nix`. Additionally, any patches should be submitted upstream before applying them in this overlay. The missing pkg-config dependency can probably be fixed in the long term by submitting a patch upstream to add it to `package.xml`.

File: distros/noetic/flatland-server/yaml_preprocessor.patch (hunk 1 above)
Comment (Educorreia932) [reply distros/noetic/flatland-server/yaml_preprocessor.patch:12]: I'm not sure about Ubuntu package, but Nix derivation seems to compile to C
https://github.com/NixOS/nixpkgs/blob/898cb2064b6e98b8c5499f37e81adbdf2925f7c5/pkgs/development/interpreters/lua-5/interpreter.nix#L85

**Comment by Educorreia932** [comment]
@lopsided98 I moved the changes to `distros/noetic/overrides.nix`

**Comment by kjeremy** [comment]
We've encountered the need to add `pkg-config` to `nativeBuildInputs` to a lot of packages for cross-compilation support.

**Pull request closed without merging**